src/devices/ecowitt.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * 55-bit one-row data packet format (inclusive ranges):
+ * |  0-14 | Always 111111101010011
+ * | 15-22 | Random(?) Sensor ID, reinitialized on boot
+ * | 23-24 | Always 00
+ * | 25-26 | 2-bit sensor channel, selectable on back of sensor {00=1, 01=2, 10=3}
+ * | 27-28 | Always 00
+ * | 29-38 | 10-bit temperature in tenths of degrees C, starting from -40C. e.g. 0=-40C
+ * | 39-46 | Always 1111 1111
+ * | 47-54 | Probably a checksum. Haven't decoded this yet. Not a simple adder, probably a CRC.
```

Comment:
Can you try with reveng or post a set of sample payloads?

Reply:
Or try with CRC-8 with poly 0x31 init 0x00
   bitbuffer_extract_bytes(bitbuffer, 0, 7, b, sizeof(b) * 8); // Skip first 7 bits

    // Verify checksum
    int crc = crc8(b, 7, 0x31, 0);

Ecowitt is some kind of brand for Fine Offset. They are probably very similar. Maybe you can merge this decoder to one of the fine offset ones.

Reply:
fineoffset_WH2_callback looks very very similar to this.

Reply:
Wow, that suggested checksum worked a treat. Nice!